e2e/terraform/provision-infra/provision-nomad/etc/nomad.d/tls.hcl

```diff
@@ -10,5 +10,5 @@ tls {
   key_file  = "/etc/nomad.d/tls/agent.key"
 
   verify_server_hostname = true
-  verify_https_client    = true
+  verify_https_client    = false // https://github.com/hashicorp/vault/issues/29435
```

Comment:
FWIW, this is the recommended value these days: https://developer.hashicorp.com/nomad/docs/concepts/security#requirements so we probably want to keep it beyond this. It might make the UI E2E tests easier too.

> Using [tls.verify_https_client=false](https://developer.hashicorp.com/nomad/docs/configuration/tls#verify_https_client) downgrades the Nomad agent HTTP server to use TLS instead of mTLS. This is safe if ACLs are enabled and is recommended if you are using the Nomad web UI to avoid the difficulty of distributing client certificates to browsers.

Although we should probably extend this bit of documentation to discuss the JWKS endpoint.

Reply:
Removed the comment within https://github.com/hashicorp/nomad/pull/25139/commits/c3ef5f1b98720cc24ebd5e8edb494a8c4ecd54e4